packages/runtime/src/store/index.ts

```diff
@@ -46,6 +47,68 @@ export class TutorialStore {
     this._previewsStore = new PreviewsStore(this._webcontainer);
     this._terminalStore = new TerminalStore(this._webcontainer, useAuth);
     this._runner = new TutorialRunner(this._webcontainer, this._terminalStore, this._stepController);
+
+    /**
+     * By having this code under `import.meta.hot`, it gets:
+     *  - ignored on server side where it shouldn't run
+     *  - discarded when doing a production build
+     */
+    if (import.meta.hot) {
+      import.meta.hot.on('tk:refresh-wc-files', async (hotFileRefs: string[]) => {
+        let shouldUpdate = false;
+
+        for (const fileRef of hotFileRefs) {
+          const result = await this._lessonFilesFetcher.invalidate(fileRef);
+
+          switch (result.type) {
+            case 'none': {
+              break;
+            }
+            case 'files': {
+              if (this._lesson?.files[0] === fileRef) {
+                shouldUpdate = true;
+
+                this._lesson.files[1] = Object.keys(result.files).sort();
+                this._lessonFiles = result.files;
+              }
```

Comment:
I think we need to break here. Otherwise we're falling through to the `solution` case as well.

Reply:
Oh good catch!!

Reply:
Whoopsie, I did!